### ISSUE TITLE
Error 0x1bc when setting wsl default version

### ISSUE THREAD

**Issue by nathanaelg** [microsoft/WSL#5651]
# Environment

```none
Windows build number: 10.0.19041.0
```

# Steps to reproduce

Attempting to update to WSL2 after updating to Windows 10 2004. Following the instructions in https://docs.microsoft.com/en-us/windows/wsl/install-win10 when I get to setting WSL2 as default: 

#  Expected behavior

# Actual behavior

I get the following error code:

```powershell
PS C:\WINDOWS\system32> wsl --set-default-version 2
Error: 0x1bc
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
```

**Comment by therealkenc** [comment]
Please collect [detailed logs](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs).

**Comment by Tealons** [comment]
I just had the same problem. Updating the WSL2 kernel (https://docs.microsoft.com/nl-nl/windows/wsl/wsl2-kernel) fixed it. 
There is information in the documentation about a possible kernel error, but it seems that the error message is somehow broken?

**Comment by oising** [comment]
Came here to report the same error (sb3 -- setting default version to 2 after having enabled vmp)

**Comment by nathanaelg** [comment]
I have tried running the kernel update but just get "This update only applies to machines with the Windows Subsystem for Linux". WSL1 still seems to be working fine.

I have collected detailed logs, should these be submitted here or through the `secure@microsoft.com`? Would they contain any private details?

**Comment by jylilaht** [comment]
> I have tried running the kernel update but just get "This update only applies to machines with the Windows Subsystem for Linux". WSL1 still seems to be working fine.
> 
> I have collected detailed logs, should these be submitted here or through the `secure@microsoft.com`? Would they contain any private details?

I have this same problem. Updated windows build, wsl2 kernel, disabled hyper-v, enabled virtualization and wsl. WSL update give s this error message and does not install.

**Comment by Tealons** [comment]
Maybe also good to mention was that some restarting was required during the process, but this is not very well documented. I found this out the hard way :)

**Comment by nathanaelg** [comment]
I have tried both restarting and uninstalling (which failed, it said it wasn't installed).

**Comment by dennis-garavsky** [comment]
>Solved my issue. You need to go right click .msi installer file and uninstall. Then install again. Like told in first note here: https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

Thank you - downloading and installing wsl_update_x64 solved this error for me.

**Comment by nathanaelg** [comment]
Detailed logs:
[lxcore.zip](https://github.com/microsoft/WSL/files/4978630/lxcore.zip)

**Comment by ageprocpp** [comment]
> Solved my issue. You need to go right click .msi installer file and uninstall. Then install again. Like told in first note here: https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

I've just solved the same error with this way. Thank you.

**Comment by lpabon** [comment]
Ok, I have resolved my issue. This is what I did and one of the following models could have resolved it:

1. Removed wsl and the vm additions
1. rebooted
1. Updated Windows 10 2004 with "Cumulative Update Preview for Windows 10..." KB4568831 (Some how it picked this up when I clicked _Check for updates_.
1. Added wsl and vm additions
1. Right clicked on `wsl_update_x64.msi` kernel to uninstall
1. rebooted
1. Installed `wsl_update_x64.msi` kernel
1. _THEN_ I typed `wsl --set-default-version 2` and it worked!

**Comment by charfole** [comment]
I also got the same issue with the 0x1bc error code. 
Hence I follow the guides in [https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel](https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel) to download and install the `wsl_update.msi` and it works.😊

**Comment by hariharan382** [comment]
for me also same problem, did all the above steps still same result

**Comment by web-assistant** [comment]
This worked for me. I went to https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel and downloaded and installed the latest WSL2 Linux kernel update package for x64 machines.

**Comment by tsubakimoto** [comment]
I had the same problem. I was able to solve it by the method written by @web-assistant .

**Comment by hariharan382** [comment]
did the same as in the link but could not update from wsl 1 to wsl2

**Comment by MallocArray** [comment]
I had the same error message which ultimately lead me here.  The Kernel update resolved it for me.

**Comment by therealkenc** [comment]
Needs-investigation as the OP submission is effectively blocked on:

>> Solved my issue. You need to go right click .msi installer file and uninstall. Then install again. Like told in first note here: https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

> I have tried both restarting and uninstalling (which failed, it said it wasn't installed).

(plus a side order of #5686)

**Comment by geekyarthurs** [comment]
> 
> 
> I just had the same problem. Updating the WSL2 kernel (https://docs.microsoft.com/nl-nl/windows/wsl/wsl2-kernel) fixed it.
> There is information in the documentation about a possible kernel error, but it seems that the error message is somehow broken?

Worked.

**Comment by headius** [comment]
Seems like we have workarounds from @lpabon and @web-assistant but this error message is obviously terrible. Perhaps the work to be done at this point is to make it say something useful, perhaps recommending the kernel update or pointing to a FAQ or walkthrough that explains the workaround?

FWIW I could not get the update to install until I uninstalled it and started over as in @lpabon comment. It kept telling me it was only for systems with WSL until I removed it and went through the full process again.

**Comment by alexhiggins732** [comment]
I had this issue on one of two machines I was installing on. After installing the virtual machine and the subsystem through both the power shell command line and windows features and rebooting, it was finally resolved following the link to install the kernel (http://aka.ms/wsl2kernel)  which was provided as error message on the second machine. That page had link to https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi to download the latest x64 kernel or https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_arm64.msi to download the latest arm kernel. After installing the appropriate x64 kernel the error went away.

**Comment by nathanaelg** [comment]
This was resolved (I believe) by a software update, I am not sure which one/s (potentially Windows Subsystem for Linux Update - 4.19.104 or 2020-08 Cumulative Update for Windows 10 Version 2004 for x64-based Systems (KB4566782)?)

**Comment by sbtcoles** [comment]
@nathanaelg  Nope

**Comment by alexhiggins732** [comment]
@nathanaelg The prior update does not show in my update history and KB4566782 is available for me to install, so neither addressed my issue, but they might prevent it in the future.

**Comment by ApfelkuchenHaus** [comment]
@nathanaelg not true. I tried this today and got the error.

**Comment by darktohka** [comment]
Still happened today, with all software updates installed.

Kernel update resolved the problem.

**Comment by mark-stopka** [comment]
> Still happened today, with all software updates installed.
> 
> Kernel update MSI resolved the problem.

Same...

**Comment by long-nguyenxuan** [comment]
> This was resolved (I believe) by a software update, I am not sure which one/s (potentially Windows Subsystem for Linux Update - 4.19.104 or 2020-08 Cumulative Update for Windows 10 Version 2004 for x64-based Systems (KB4566782)?)

System up to date at 2020-08-17 07:31 but it still error :'( 
![image](https://user-images.githubusercontent.com/46129500/90347614-c16ccc00-e05b-11ea-9f59-34d1d1c89cc3.png)

**Comment by long-nguyenxuan** [comment]
> > This was resolved (I believe) by a software update, I am not sure which one/s (potentially Windows Subsystem for Linux Update - 4.19.104 or 2020-08 Cumulative Update for Windows 10 Version 2004 for x64-based Systems (KB4566782)?)
> 
> System up to date at 2020-08-17 07:31 but it still error :'(

update:
restart and reinstall wsl2 kernel update make it works, magic!! 😂

**Comment by drk-mtr** [comment]
Just experienced this on a brand new PC with a fresh Windows install, on version 2004, with updates checked today 19/08/2020. the resolution was as described i.e. install the kernel update - thanks for sharing the fix.

**Comment by vinsent2002** [comment]
Why should I install an add-on package for WSL 2 to work?

**Comment by long-nguyenxuan** [comment]
> Why should I install an add-on package for WSL 2 to work?

because it's **2**, and it requires an add-on package to work, if you don't want to install addons package, feel free to stay with **1**

**Comment by gahstanz** [comment]
Hey guys,

After installing the wsl2-kernel a bit too hasty, I was receiving the same error mentioned. I was instructed to install the kernel update but received the "_**This update only applies to machines with the Windows Subsystem for Linux**_" error. If you've already done this like me then I'd assume you've already downloaded the kernel update and done the same.

I was able to resolve this by doing the following:

1. Download the WSL 2 Kernel Update from the following link (this will download wsl_update_x64.msi):
https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

2. If you try to run this, you'll get the WSL error as mentioned above. Right click on the downloaded wsl_update_x64.msi, select "**Uninstall**".

3. Once completed, I reinstalled the wsl_update_x64.msi successully and no without receiving the following error "**_This update only applies to machines with the Windows Subsystem for Linux_**".

4. Open Powershell as Administrator, and set the version to WSL 2 by typing in the following.
`wsl --set-default-version 2`

You should no longer get the error.
![image](https://user-images.githubusercontent.com/5556069/91508889-d8010600-e92c-11ea-8ba8-3b30718ea5c5.png)

**Comment by ThermoMan** [comment]
The msi installer fixed my problem too.  No other fix steps were required.

**Comment by saad-gh** [comment]
> I just had the same problem. Updating the WSL2 kernel (https://docs.microsoft.com/nl-nl/windows/wsl/wsl2-kernel) fixed it.
> There is information in the documentation about a possible kernel error, but it seems that the error message is somehow broken?

This fix worked for me as well

**Comment by ruant** [comment]
Why is the kernel installation poping up as `Publisher: Unknown`?
Isn't this a Microsoft made file? (Signature says so..)

Maybe it's just my computer and it's CA that's borked?

**Comment by therealkenc** [comment]
> Maybe it's just my computer and it's CA that's borked?

That's #5628, you aren't alone. And yes it's weird (and remains unexplained) because the file itself appears kosher.

**Comment by ManoloMoraes** [comment]
> Solved my issue. You need to go right click .msi installer file and uninstall. Then install again. Like told in first note here: https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

Thank you, uninstall and install wsl_update_x64.msi solved my first problem, other problem solved enable  virtualization on BIOS.

**Comment by amitai** [comment]
The above suggestions did not fix my issue! :( I am still having a problem.

Like others above, I attemped to uninstall wsl_update_x64.msi. However, I get the error: 

> This action is only valid for products that are currently installed.

According to [the FAQ on this error](https://docs.microsoft.com/nl-nl/windows/wsl/wsl2-kernel#this-update-only-applies-to-machines-with-the-windows-subsystem-for-linux) it could be one of these:

1. My Windows version not supporting WSL2? Not the case: Windows 10 v2004 build 19041.450.
2. WSL not enabled? Not the case: I see it is checked in turn features on or off. When I run wsl.exe it says I have no distros installed... but the command clearly runs (and I can't install any distros because it says I need the kernel update)
3. Reboot required? I rebooted.

**Comment by moditpro365** [comment]
Help! I have uninstalled and reinstalled WSL components. The kernel update appears to install successfully. But when I try to set version 2 as default it tells me that the update is required. I have uninstalled and reinstalled the update. I get no errors. I've rebooted multiple times. Any assistance would be greatly appreciated. I'm racking my brain.
